File: src/UI/SDK/src/models/HSOrderReturn.ts
import { OrderReturn } from 'ordercloud-javascript-sdk'

export interface HSOrderReturn extends OrderReturn<OrderReturnXp> {}

export interface OrderReturnXp {
    SellerComments?: string
    SubmittedStatusDetails?: ReturnEventDetails
    ApprovedStatusDetails?: ReturnEventDetails
    DeclinedStatusDetails?: ReturnEventDetails
    CompletedStatusDetails?: ReturnEventDetails
    CanceledStatusDetails?: ReturnEventDetails
}

export interface ReturnEventDetails {
    ProcessedByName?: string
    ProcessedByUserId?: string
    ProcessedByCompanyId?: string // empty if Admin, otherwise SupplierID if supplier and BuyerID if buyer
    RefundAmount?: number
}

File: src/Middleware/src/Headstart.Common/Models/Headstart/HSOrderReturn.cs
using System;
using System.Collections.Generic;
using System.Text;
using OrderCloud.SDK;

namespace Headstart.Common.Models
{
    public class HSOrderReturn : OrderReturn<OrderReturnXp>
    {
    }

    public class OrderReturnXp
    {
        public string SellerComments { get; set; }

        public ReturnEventDetails SubmittedStatusDetails { get; set; }

        public ReturnEventDetails ApprovedStatusDetails { get; set; }

        public ReturnEventDetails DeclinedStatusDetails { get; set; }

        public ReturnEventDetails CompletedStatusDetails { get; set; }

        public ReturnEventDetails CanceledStatusDetails { get; set; }
    }

    public class ReturnEventDetails
    {
        public string ProcessedByName { get; set; }

        public string ProcessedByUserId { get; set; }

        public string ProcessedByCompanyId { get; set; }

        public decimal RefundAmount { get; set; }
    }
}
